### ISSUE TITLE
Support Postgres's array type

### ISSUE THREAD

**Issue by kevinburkeshyp** [balderdashy/sails-postgresql#195]
Currently Waterline columns with type `'array'` are created as a TEXT type in the database, and written/serialized using JSON.stringify/JSON.parse. This behavior is unintuitive, and doesn't support, for example, checking list membership without pulling the entire array into memory and searching it.

I understand this would be a lot of work to implement, but it seems like a valid feature request (and a placeholder ticket for people who are curious about this feature and visiting from Google).

Duplicate of #36, which was closed.

**Comment by devinivy** [comment]
Totally valid feature request!  This might be considered a "breaking feature" though, since it affects the created and expected database schema.  Thanks for reopening :)

**Comment by tjwebb** [comment]
@kevinburkeshyp I've supported it in my es6 re-write; https://github.com/waterlinejs/postgresql-adapter/blob/master/lib/util.js#L227-L228. It's  fairly nascent, so there may be issues.

**Comment by tjwebb** [comment]
> This might be considered a "breaking feature" though, since it affects the created and expected database schema.

I think breaking in terms of upgrading one pg adapter version to another, but not against the Waterline spec.

**Comment by sailsbot** [comment]
Thanks for posting, @kevinburkeshyp.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
